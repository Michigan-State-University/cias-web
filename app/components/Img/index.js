import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import { memo } from 'react';
import { layout, margin, style, positioning } from '../BaseComponentStyles';

const Img = styled.img.attrs((props) => ({
  onClick: props.disabled ? null : props.onClick,
}))`
  width: auto;
  height: auto;
  pointer-events: ${({ pointerEvents }) => pointerEvents || ''};
  ${layout};
  ${margin};
  ${style};
  ${positioning};
  ${({ flipHorizontal }) =>
    flipHorizontal &&
    css`
      -moz-transform: scaleX(-1);
      -webkit-transform: scaleX(-1);
      -o-transform: scaleX(-1);
      transform: scaleX(-1);
    `}
`;

Img.propTypes = {
  pointerEvents: PropTypes.string,
  flipHorizontal: PropTypes.bool,
};

export default memo(Img);
