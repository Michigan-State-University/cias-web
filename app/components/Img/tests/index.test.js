import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import Img from '../index';

describe('<Img />', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Img />);
    expect(container).toMatchSnapshot();
  });
  it('should match disabled snapshot', () => {
    const { container } = render(<Img disabled />);
    expect(container).toMatchSnapshot();
  });
});
