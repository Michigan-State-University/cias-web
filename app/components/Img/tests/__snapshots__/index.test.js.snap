// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Img /> should match disabled snapshot 1`] = `
.c0 {
  width: auto;
  height: auto;
  opacity: 0.5;
}

<div>
  <img
    class="c0"
    disabled=""
  />
</div>
`;

exports[`<Img /> should match the snapshot 1`] = `
.c0 {
  width: auto;
  height: auto;
}

<div>
  <img
    class="c0"
  />
</div>
`;
